igvc_navigation/src/navigation_server/navigation_server.cpp

```diff
@@ -173,52 +174,82 @@ void NavigationServer::actionGetPathDone(const actionlib::SimpleClientGoalState
       break;
   }
 
-  if (navigation_state_ != EXE_PATH)
+  // only start replanning if we are executing path
+  if (navigation_state_ == EXE_PATH)
   {
-    return;
+    current_replanning_tries_ = 0;
+    // prevent other threads from replanning while thread sleeps
```

Comment:
Why are there multiple threads trying to replan?

Reply:
I think the reason they have the mutex is because there two points where replanning is done: are the end of actionGetPathDone and actionGetPathReplanningDone. The mutex prevents both from executing at the same time, which could happen if the robot finishing replanning just as it gets a new navigation goal.